### ISSUE TITLE
Android: Copy resource files in the correct folder

### ISSUE THREAD

**Issue by GitToTheHub** [apache/cordova-android#1753]
# Bug Report
I want to copy resources to the `res` directory of the android app.

## Problem
I followed the documentation for the [resource-file](https://cordova.apache.org/docs/en/12.x/config_ref/#resource-file) tag of the `config.xml`, where it shows an example to copy a resource file to `res/values/` for Android:
```xml
<resource-file src="FooPluginStrings.xml" target="res/values/FooPluginStrings.xml" />
```
When I use this code, the resource file gets copied to `platforms/android/res/values`, but the correct folder would be `platforms/android/app/src/main/res/values`.

When I do the same for iOS with

```xml
<platform name="ios">
    <resource-file src="FooPluginStrings.xml" target="FooPluginStrings.xml" />
</platform>
```

It gets correctly copied to `platforms/iOS/[AppName]/Resources` (using cordova-ios 7.1.1).

Should there be not a similar behavior for Android? Must the documentation be updated to the correct path, or is this an issue in the Cordova-cli?

### What is expected to happen?
The resource file should be put in the correct folder like the documentation says.

### What does actually happen?
The resource file is copied in the wrong place.

### Version information

- macOS 15.2
- MacBook Pro 2019 13"
- cordova-cli 12.0.0 (cordova-lib@12.0.2)

## Checklist

- [ ] I searched for existing GitHub issues
- [ ] I updated all Cordova tooling to most recent version
- [ ] I included all the necessary information above

**Comment by breautek** [comment]
> When I use this code, the resource file gets copied to platforms/android/res/values, 

Android has a concept of build variants so i don't think it's a good idea to auto prefix and assume app/src/main.

`app` here is the module name, and `main` is the build type. Complex projects could make use of these.

So I think we should consider this a doc issue.

**Comment by GitToTheHub** [comment]
Thank you @breautek, I didn't know about this. Could this path nevertheless automatically calculated for build variants? Is it really needed do decide in which build variant a resource file will be copied? Does `cordova-android` support build variants anyway?

**Comment by breautek** [comment]
> Could this path nevertheless automatically calculated for build variants?

I don't think so. Android has a few reserved ones, `main`, `androidTest` (for connected android device/simulator testing), and `test` (unit tests that runs in a standard JVM, for code that doesn't depend on android SDKs). `main` is the default/fallback. All other build variants will get merged with `main`'s assets and configurations.

So outside of those 3, the build variant name is entirely up to the developer. Cordova CLI only ever uses `main` and doesn't make use of build variants, not even the test variants, but it doesn't mean that complex projects adds their own build variants and projects using `resource-file` would assume current behaviour.

Even if we conditionally prefix and assume `main` build variant there is going to be a lot of ambiguity as android assets are split into several folders, depending on the purpose, for example:

- res/values/x.xml
- res/xml/x.xml
- res/layouts/x.xml
- res/drawable/x.xml

And each one of those directories can have variants as well, for localization, themes, device form factors, etc:

- res/values-fr/x.xml (for french)
- res/values-night/x.xml (for dark mode)

And each of these variants can be compounded, for example: `res/values-fr-night/x.xml` for dark mode, french locale device configuration, or `res/values-fr-night-v24/x.xml` for french, dark mode, on API 24+ devices. Android about 20 different variant types, so I trust you can see how this can be problematic.

I think if we WERE going to do any sort of prefixing, the path must begin with `/res/...` (or `res/...`) then we can assume `app/src/main/` should be prefixed. Otherwise we should attempt to use the path as is. The resource type (`xml`, `drawable`, `values`, etc) must be explicitly defined along with the resource file name.

```xml
<resource-file src="Foo.xml" target="res/values/Foo.xml" /> <!-- goes to app/src/main/res/values/Foo.xml -->
<resource-file src="Foo.xml" target="/res/values/Foo.xml" /> <!-- goes to app/src/main/res/values/Foo.xml -->
<resource-file src="Foo.xml" target="app/src/main/res/xml/Foo.xml" /> <!-- goes to app/src/main/res/xml/Foo.xml -->
<resource-file src="Foo.xml" target="/bar/res/values/Foo.xml" /> <!-- goes to bar/res/values/Foo.xml which is not really a valid location -->
<resource-file src="Foo.xml" target="app/src/bar/res/values/Foo.xml" /> <!-- goes to app/src/bar/res/values/Foo.xml -->
<resource-file src="Foo.xml" target="foobar/src/main/res/values/Foo.xml" /> <!-- goes to foobar/src/main/res/values/Foo.xml which is valid only if something has created the foobar android module -->
```

The key point here is we should be careful not to prefix something seems be an absolute path to a particular android module and/or build variant.

> Does cordova-android support build variants anyway?

We don't have any explicit directives or anything to create build variants but it can be done via `<config-file>` or `<edit-config>` or hooks, etc.

**Comment by GitToTheHub** [comment]
Hi @breautek,

thank you for your very detailed answer and explanation.

I know about these resource identifiers and my thought was also that the path should start with `res`. After this, the user can define, whatever he want.

When Cordova is used for building or preparing, I think it should know where the destination, otherwise it could not build/prepare? So if Cordova knows it builds to `android/app` it can assume that `res/[resource-path]`must be copied to `android/app/res/[resource-path]`. If Cordova is not used for building, then `resource-file` could not be used. If it is possible to build or prepare with Cordova to another build variant, other then `main`, it should know where to copy resource files?
